src/Raven.Server/Documents/Indexes/Static/Roslyn/Rewriters/MethodDetectorRewriter.cs

```diff
@@ -68,5 +77,16 @@ public override SyntaxNode VisitGroupClause(GroupClauseSyntax node)
             Methods.HasGroupBy = true;
             return base.VisitGroupClause(node);
         }
+
+        private static void AssertLoadDocumentHasConstantCollectionName(InvocationExpressionSyntax node)
+        {
+            if (node.ArgumentList.Arguments.Count == 2)
```

Comment:
Is it possible that this might _not_ be the case?

Reply:
No, we always have two arguments (id and collection name). For `LoadDocument` call without collection name parameter we add it in `ExpressionStringBuilder`. I removed this check